demo/src/index.js

```diff
@@ -36,6 +39,51 @@ ${clauseText}
   return slateTransformer.fromMarkdown(defaultContractMarkdown);
 };
 
+/**
+ * Parses user inputted text for a template using Cicero
+ * @param {object} template The cicero template object.
+ * @param {string} text The user submitted text.
+ * @param {string} clauseId The uuid of the clause
+ * @returns {Promise} The result of the parse or an error.
+ */
+const parseClause = (template, text, clauseId) => {
+  try {
+    const ciceroClause = new Clause(template);
+    ciceroClause.parse(text);
+    const parseResult = ciceroClause.getData();
+    // Example code you could use to add this parse result to a redux store:
+    // store.dispatch(parseClauseSuccess(clauseId, parseResult));
+    return Promise.resolve(parseResult);
+  } catch (error) {
+    // Example code you could use to add this parse result to a redux store:
+    // store.dispatch(parseClauseError(clauseId, error));
+    return Promise.reject(error);
+  }
+};
+
+const fetchTemplateObj = async (action) => {
+  try {
+    const templateObj = await Template.fromUrl(action.uri);
+    return templateObj;
+  } catch (err) { return err; }
+};
+
+/**
+ * Called by the clause plugin into the contract editor
+ * when we need to load a template
+ * Note: This is used for example and is not actually used in the demo
+ */
+const loadTemplate = async (templateUri) => {
```

Comment:
Can we make this simpler by eliminating this `loadTemplate` method and instead passing the `fetchTemplateObj` method as the `loadTemplateObject` prop?

Reply:
Yes, I kept this as an example, but maybe that's just overly complicated.